x-pack/legacy/plugins/alerting/server/init.ts

```diff
@@ -17,50 +18,89 @@ import {
 } from './routes';
 import { AlertingPlugin, Services } from './types';
 import { AlertTypeRegistry } from './alert_type_registry';
-import { AlertsClient } from './alerts_client';
+import { AlertsClient, CreateAPIKeyResult } from './alerts_client';
 import { SpacesPlugin } from '../../spaces';
+import { KibanaRequest } from '../../../../../src/core/server';
+import { PluginSetupContract as SecurityPluginSetupContract } from '../../../../plugins/security/server';
 import { createOptionalPlugin } from '../../../server/lib/optional_plugin';
 
 export function init(server: Legacy.Server) {
   const config = server.config();
+  const kbnServer = (server as unknown) as KbnServer;
+  const taskManager = server.plugins.task_manager!;
+  const { callWithRequest } = server.plugins.elasticsearch.getCluster('admin');
   const spaces = createOptionalPlugin<SpacesPlugin>(
     config,
     'xpack.spaces',
     server.plugins,
     'spaces'
   );
-
-  const { callWithInternalUser } = server.plugins.elasticsearch.getCluster('admin');
-  const savedObjectsRepositoryWithInternalUser = server.savedObjects.getSavedObjectsRepository(
-    callWithInternalUser
+  const security = createOptionalPlugin<SecurityPluginSetupContract>(
+    config,
+    'xpack.security',
+    kbnServer.newPlatform.setup.plugins,
+    'security'
   );
 
-  function getServices(basePath: string): Services {
-    const fakeRequest: any = {
-      headers: {},
-      getBasePath: () => basePath,
-    };
+  server.plugins.xpack_main.registerFeature({
+    id: 'alerting',
+    name: 'Alerting',
+    app: ['alerting', 'kibana'],
+    privileges: {
+      all: {
+        savedObject: {
+          all: ['alert'],
+          read: [],
+        },
+        ui: [],
+      },
+      read: {
+        savedObject: {
+          all: [],
+          read: ['alert'],
+        },
+        ui: [],
+      },
+    },
+  });
+
+  // Encrypted attributes
+  server.plugins.encrypted_saved_objects!.registerType({
+    type: 'alert',
+    attributesToEncrypt: new Set(['apiKey']),
+    attributesToExcludeFromAAD: new Set([
```

Comment:
They were mostly left out because of these values can be updated and would break the AAD / alert. From my understanding of the concern, we should generate a new API key on update from the user in context?

Reply:
> They were mostly left out because of these values can be updated and would break the AAD / alert. From my understanding of the concern, we should generate a new API key on update from the user in context?

Leaving these values out of the AAD allows them to be updated without having to generate a new API Key on update.

Reply:
Fixed in https://github.com/elastic/kibana/pull/41389/commits/b1a40d576b2c4bc86364f2510250acfbe1b96587.

Reply:
I have created a card in project backlog to cleanup the old API key on update https://github.com/elastic/kibana/projects/26#card-25354998. This will require a PR to add remove API key support in the security plugin and then find a way to delete the key without the owner being in context / `request`.